internal/runtime/client/fake/fake_client.go

```diff
@@ -68,30 +68,37 @@ func (f *RuntimeClientBuilder) MarkReady(ready bool) *RuntimeClientBuilder {
 }
 
 // Build returns the fake runtime client.
-func (f *RuntimeClientBuilder) Build() runtimeclient.Client {
-	return &runtimeClient{
-		isReady:          f.ready,
-		callAllResponses: f.callAllResponses,
-		callResponses:    f.callResponses,
-		catalog:          f.catalog,
+func (f *RuntimeClientBuilder) Build() *RuntimeClient {
+	return &RuntimeClient{
+		isReady:                  f.ready,
+		callAllResponses:         f.callAllResponses,
+		callResponses:            f.callResponses,
+		catalog:                  f.catalog,
+		callAllExtensionsTracker: map[runtimecatalog.GroupVersionHook]int{},
 	}
 }
 
-var _ runtimeclient.Client = &runtimeClient{}
+var _ runtimeclient.Client = &RuntimeClient{}
 
-type runtimeClient struct {
+// RuntimeClient is a fake implementation of runtimeclient.Client.
+type RuntimeClient struct {
 	isReady          bool
 	catalog          *runtimecatalog.Catalog
 	callAllResponses map[runtimecatalog.GroupVersionHook]runtimehooksv1.ResponseObject
 	callResponses    map[string]runtimehooksv1.ResponseObject
+
+	callAllExtensionsTracker map[runtimecatalog.GroupVersionHook]int
 }
 
 // CallAllExtensions implements Client.
-func (fc *runtimeClient) CallAllExtensions(ctx context.Context, hook runtimecatalog.Hook, request runtime.Object, response runtimehooksv1.ResponseObject) error {
+func (fc *RuntimeClient) CallAllExtensions(ctx context.Context, hook runtimecatalog.Hook, request runtime.Object, response runtimehooksv1.ResponseObject) error {
 	gvh, err := fc.catalog.GroupVersionHook(hook)
 	if err != nil {
 		return errors.Wrap(err, "failed to compute GVH")
 	}
+	defer func() {
```

Comment:
The implementation here used GVH as the key and hence I couldn't put it at the beginning of the function. Switched to using the hookName for the key. Now it can moved to the beginning. Also using hookName seems like a better approach. Less dependency.